server/engine/engine.go

```diff
@@ -292,3 +298,10 @@ func (engine *Engine) runStep(step model.Step, execution *model.Execution, waitG
 	model.UpdateExecution(execution, deployResponse, "")
```

Comment:
I saw the same thing during my testing and when I looked where it came from, `server/engine/engine.go:284`, I concluded it is ok because the engine has one go routine waiting on the deployment to finish and then another one calls cancel on the deployment, causing the first one to report a failure.
Here is that line:
```go
deployResponse, err := azure.WaitForDeployARMTemplate(engine.context, step.Name, deployment)
```
In separate PR we can update the code in engine to check the status code first, and then print a message that's appropriate for status of "Canceled".

Reply:
Yeah no problem!